src/core_plugins/elasticsearch/lib/clusters.js

```diff
@@ -0,0 +1,14 @@
+import Cluster from './cluster';
+
+const clusters = new Map();
```

Comment:
@tylersmalley Attaching the state to a runtime object like server means the logic for getCluster and createCluster can be used independently of the state.  The most immediate benefit of that is for testing, since the state becomes ephemeral and you can disregard it/start from scratch at will rather than worrying about cleaning things up in an afterEach (which, by the way, you wouldn't even be able to do properly in the current implementation since there's no way to remove clusters from the map).

As an alternative, this could just be a class or factory that we could invoke once during plugin init and use elsewhere.  To be honest, I'm not even sure how feasible this is in the current plugin model, though, so it might not even be in the cards.